### ISSUE TITLE
dotted-quad support for IPv4 subnet masks

### ISSUE THREAD

**Issue by abraxxa** [maxmind/Net-Works#30]
I have to work with a product that returns IPv4 networks with the subnet mask in dotted-quad notation and need to parse those.
Can we add support for in- and output of this format without breaking things?

Thanks!

**Comment by oschwald** [comment]
That seems possible. A PR that did this with appropriate tests and documentation would likely be accepted.

**Comment by abraxxa** [comment]
I was already able to use use Net::Works::Util::_integer_address_to_string($net->_subnet_integer) for generting the dotted-quad output.
The guesswork in new_from_string is what worries me.

**Comment by oschwald** [comment]
Yeah, I could see guessing potentially leading to confusion in some cases. I suppose an additional parameter, e.g., `dotted_decimal_mask => 1` or something could work.

**Comment by abraxxa** [comment]
@oschwald Before I start with this pull-request, can you be so kind to look at my fork which uses binary representation instead of integer/Math::Int128?
My goal was to get rid of Math::Int128 as far as I can remember and provide an API for binary values to store those in a SQL database that has no Int128 or IP address/network support (Oracle) using its RAW database which IO is in uppercase hex.
Thanks!

**Comment by abraxxa** [comment]
I've rebased my fork with my commits from 2014 against current main and called it main, so my original modifications are also viewable in the still existing master branch.

**Comment by oschwald** [comment]
@abraxxa, sure, do you have a link to what you want me to review?

**Comment by abraxxa** [comment]
The main branch in my fork please.

**Comment by abraxxa** [comment]
@oschwald friendly reminder ;-)

**Comment by oschwald** [comment]
Sorry for the delay. At a high level, most of the changes seem good. It will be easier to review once there is a PR that I can comment on.

**Comment by oschwald** [comment]
We are no longer actively maintaining this distribution. I am closing this issue as the repo will be archived.
